mitodl/ol-keycloak
# Style email templates to match mockups

### What are the relevant tickets?




Closes #48 

### Description (What does it do?)


Adds email templates for the following:

- A generic email base template. This is a catch-all for any emails Keycloak sends not themed specifically.
- Reset Password email
- Confirm email (new account)

### Screenshots (if appropriate):
![aNe8z5usdpjXTLCX3A6DEa](https://github.com/mitodl/ol-keycloak/assets/28598/78934460-c818-451c-9d9c-69bb10f8be8c)
![NtP7KYn5mVdEVdbWfVyPbD](https://github.com/mitodl/ol-keycloak/assets/28598/8a4fc9b7-f3c5-47e5-8371-1013cfee12ab)

### How can this be tested?

- Go to Realm Settings -> General and ensure that your realm has a display name "MIT Open", otherwise you may see a different name in the emails.
- Go to Realm Settings -> Email and at the bottom of the page send a test email. This should use the catch-all email template.
- Register a new user, you should see the themed email.
- Reset your password, you should see the themed email.

### Additional Context


Writing HTML emails that look good across all email clients is a headache. In the past dealing with this has been to copy/paste a static template someone has put up somewhere, but for this effort I pulled in the tool [mjml](https://documentation.mjml.io/), which gives you a very lightweight markup syntax that the tool transforms into a full html email. This has the tradeoff of not being able to nested FTL templates, because the tool can't convert partial files, but I think that's a net win. This tool appears to be the industry standard these days for generating well styled html emails and is maintained by the email marketing platform Mailjet so I don't think it's going anywhere anytime soon.

I also added a github action that verifies there's no changes to the MJML templates that haven't been compiled and committed.
